src/engine/strat_engine/filesystem.rs

```diff
@@ -109,6 +169,30 @@ impl Recordable<FilesystemSave> for StratFilesystem {
     }
 }
 
+/// Return total bytes, total bytes used
+pub fn fs_usage(mount_point: &PathBuf) -> EngineResult<(Bytes, Bytes, Bytes)> {
+    let mut stat = Statvfs::default();
+    try!(statvfs(mount_point, &mut stat));
+    let total_bytes = stat.f_blocks * stat.f_bsize;
+    Ok((Bytes(total_bytes), Bytes(total_bytes - stat.f_bfree * stat.f_bsize), Bytes(stat.f_bsize)))
+}
+
+/// Use the xfs_growfs command to expand a filesystem mounted at the given
+/// mount point.
+pub fn xfs_growfs(mount_point: &Path, expand_size: DataBlocks) -> EngineResult<()> {
+    if try!(Command::new("xfs_growfs")
+                .arg(mount_point)
+                .arg("-D")
+                .arg(format!("{}", *expand_size))
```

Comment:
We should also be aware that we don't necessarily get what we ask for when it comes to expanding sizes.